osu.Game/Skinning/LegacySkin.cs

```diff
@@ -331,6 +336,8 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                     switch (target.Target)
                     {
                         case SkinnableTarget.MainHUDComponents:
+                            if (!this.HasFont(LegacyFont.Score) && !AllowDefaultHUDComponentsFallback)
```

Comment:
I don't understand this. Can you explain what this is doing?

Reply:
If the current skin does not have the required resource, it must return `null` instead to fall back to a parent skin, otherwise, most child skin (`LegacyBeatmapSkin` in this case) will return its own HUD components instead.

And this needs the `AllowDefaultHUDComponentsFallback` check to allow legacy skin to fall back to default lazer skin HUD components temporarily while disallowing the legacy beatmap skin from doing so.

Hopefully, this will improve when supporting falling back to default legacy skin instead, `AllowDefaultHUDComponentsFallback` wouldn't need to exist anymore.

Reply:
That makes the `AllowDefaultHUDComponentsFallback` thing backwards, no? Reading your statement again, you say

> If the current skin does not have the required resource, it must return `null` instead to fall back to a parent skin

but then, the code can only ever return `null` if that property is _false_. Because it's negated in the condition. So fallback (achieved by returning `null`) can only ever happen if `!AllowDefaultHUDComponentsFallback`. Am I missing something?

Reply:
Should make more sense with the alternative way now, and will all go away with legacy default skin fallback anyways.